cmd/tkey-ssh-agent/pinentry.go

```diff
@@ -108,3 +120,39 @@ func findWindowsPinentry() string {
 
 	return found
 }
+
+var macOSScriptTemplate = template.Must(template.New("script").Parse(`
+var app = Application.currentApplication()
+app.includeStandardAdditions = true
+app.displayDialog(
+	"{{ .Message }}", {
+    defaultAnswer: "",
+	withTitle: "{{ .Title }}",
+    buttons: ["Cancel", "OK"],
+    defaultButton: "OK",
+	cancelButton: "Cancel",
+    hiddenAnswer: true,
+})`))
+
+func macOSPrompt(msg, title string) (string, error) {
+	script := new(bytes.Buffer)
+	if err := macOSScriptTemplate.Execute(script, map[string]interface{}{
+		"Message": strings.ReplaceAll(msg, "\n", `\n`), "Title": title,
+	}); err != nil {
+		return "", err
+	}
+
+	c := exec.Command("osascript", "-s", "se", "-l", "JavaScript")
+	c.Stdin = script
+	out, err := c.Output()
+	if err != nil {
+		return "", fmt.Errorf("failed to execute osascript: %v", err)
```

Comment:
I have one question, is there a possibility to distinguish between different return values from the osascript we invoke?

If I press "cancel" I will receive this error:
```
Failed to load app: Failed to get USS: macOS Prompt: failed to execute osascript: exit status 1
```
Running the same JavaScript in the Script Editor I do get
```
Error -128: User cancelled.
```
and from the AppleScript Language Guide it seems to be a defined error.
But from what I can tell it seems that all errors simply becomes "exit status 1" in `C.Output()`

Ideally it would be great to distinguish between a user cancellation and other errors. Do you know it is possible?